File: |-- src_R/|   |-- 03_akar_kuadrat.R
# PROGRAM 3: AKAR PERSAMAAN KUADRAT
# FUNGSI
hitung_akar <- function(a, b, c) {
  # Validasi: a tidak boleh 0
  if (a == 0) {
    cat("STATUS: TIDAK VALID - Bukan persamaan kuadrat (a = 0)\n")
    return()
  }
  # Hitung diskriminan
  D <- b^2 - 4 * a * c
  
  if (D > 0) {    # Kasus 1: D > 0 (dua akar real berbeda)
    x1 <- (-b + sqrt(D)) / (2 * a)
    x2 <- (-b - sqrt(D)) / (2 * a)
    cat("STATUS: VALID - Akar real berbeda\n")
    cat("Diskriminan (D) =", D, "\n")
    cat("x1 =", sprintf("%.3f", x1), "\n")
    cat("x2 =", sprintf("%.3f", x2), "\n")
  } else if (D == 0) {    # Kasus 2: D == 0 (akar real kembar)
    x <- -b / (2 * a)
    cat("STATUS: VALID - Akar real kembar\n")
    cat("Diskriminan (D) =", D, "\n")
    cat("x1 = x2 =", sprintf("%.3f", x), "\n")
  } else {   # Kasus 3: D < 0 (akar imajiner)
    cat("STATUS: VALID - Akar imajiner\n")
    cat("Diskriminan (D) =", D, "\n")
    cat("Persamaan hanya memiliki akar imajiner\n")
  }
}
# SKENARIO 1 (Akar real berbeda)
hitung_akar(1, -5, 6)

# SKENARIO 2 (Akar real kembar)
hitung_akar(1, -4, 4)

# SKENARIO 3 (Akar imajiner)
hitung_akar(1, 2, 5)

#SKENARIO 4 (Akar Imajiner)
hitung_akar(3,5,7)

File: |-- src_python/|   |-- 03_akar_kuadrat.py
#PROGRAM 3: AKAR PERSAMAAN KUADRAT

import math

def hitung_akar(a, b, c):
    # Hitung diskriminan
    D = b**2 - 4 * a * c
    
    if D > 0:
        x1 = (-b + math.sqrt(D)) / (2 * a)
        x2 = (-b - math.sqrt(D)) / (2 * a)
        print("STATUS: VALID - Akar real berbeda")
        print(f"Diskriminan (D) = {D}")
        print(f"x1 = {x1:.3f}")
        print(f"x2 = {x2:.3f}")
    
    elif D == 0:
        x = -b / (2 * a)
        print("STATUS: VALID - Akar real kembar")
        print(f"Diskriminan (D) = {D}")
        print(f"x1 = x2 = {x:.3f}")
    
    else:
        print("STATUS: VALID - Akar imajiner")
        print(f"Diskriminan (D) = {D}")
        print("Persamaan hanya memiliki akar imajiner")


# SKENARIO 1 
print("SKENARIO 1")
hitung_akar(1, -5, 6)

# SKENARIO 2 
print("SKENARIO 2")
hitung_akar(1, -4, 4)

# SKENARIO 3 
print("SKENARIO 3")
hitung_akar(1, 2, 5)

# SKENARIO 4
print("SKENARIO 4")
hitung_akar(3, 5, 7)
